app/controllers/comments_controller.rb

```diff
@@ -1,31 +1,52 @@
 class CommentsController < ApplicationController
 
-  http_basic_authenticate_with name: "sebas", password: "secret", only: :destroy
+  include PrivateGuard
+  include SecureTransaction
+
+  before_action :require_user
 
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.create(comment_params)
-    redirect_to article_path(@article)
+    if (@article = find(Article, params[:article_id])).is_a?(Article)
```

Comment:
@JsMontesR again, too much logic that should not be placed in a controller, as a tip: "Controllers should be just in charge of invoking "operations" and rendering "templates"